### PR TITLE
Add configure action for all-in-one

### PR DESCRIPTION
**Pull request by javacruft**
Add a configure action which prompts the user for a set of data to configure a) a user, project and networking for use of the cloud and b) external network information for the cloud for north/south traffic.

This is applied using a terraform configuration.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -19,6 +23,14 @@ apps:
       - network-bind
       - ssh-public-keys
 
+  terraform:
+    command: bin/terraform
+    plugs:
+      - home
+      - network
+      - ssh-public-keys
+      - juju-client-observe
```
Comment (hemanthnakkina) [inline snap/snapcraft.yaml:32]: Is this required? Or is it for future use when juju module is available for terraform?

**Comment by hemanthnakkina** [comment]
Can we add terraform log with default to INFO to the location $SNAP_COMMON/terraform/tf_{datetime}.log
Required env variables:
TF_LOG
TF_LOG_PATH

This can be separate PR as well.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -35,6 +47,32 @@ parts:
...
+    stage:
+      - bin/terraform
+
+  configure:
+    plugin: dump
+    source: configure/
+    override-build: |
+      snapcraftctl build
+      cd $SNAPCRAFT_PART_INSTALL/etc/configure
+      $SNAPCRAFT_PRIME/bin/terraform providers mirror $SNAPCRAFT_PRIME/terraform-providers
```
Comment (wolsen) [inline snap/snapcraft.yaml]: I think we actually also need the terraform providers that we care about to be built with the snap as they are not built for all platforms sadly. I believe only `amd64` and `arm` (honestly not sure of the flavor here) are published by hashicorp today.

**Review by wolsen — APPROVED** [review]

**Review by gnuoy — APPROVED** [review]
Seems reasonable to me

**Comment by wolsen** [comment]
> Can we add terraform log with default to INFO to the location $SNAP_COMMON/terraform/tf_{datetime}.log Required env variables: TF_LOG TF_LOG_PATH
> 
> This can be separate PR as well.

FYI, I've discussed this with @javacruft  and decided to do this as a separate PR. It is a very needed PR though.

**Pull request merged**